extensions/wikia/PortableInfobox/tests/PortableInfoboxMobileRenderServiceTest.php

```diff
@@ -0,0 +1,621 @@
+<?php
+
+class PortableInfoboxMobileRenderServiceTest extends WikiaBaseTest {
+	//todo: https://wikia-inc.atlassian.net/browse/DAT-3076
+	//todo: we are testing a lot of functionality and have issues with mocking
+	//todo: we should move all render service test to API tests
+
+	protected function setUp() {
+		$this->setupFile = dirname( __FILE__ ) . '/../PortableInfobox.setup.php';
+		parent::setUp();
+	}
+
+	/**
+	 * @param $input to check presence of some additional config fields. Possible fields:
+	 * 'isInvalidImage' - bool - if getThumbnail should return false
+	 * 'isMercury' - bool - if we want to test Mercury skin
+	 * 'isMercuryExperimentalMarkupEnabled' - bool
+	 * 'smallImageDimensions' - integer - size of small image (both width and height)
+	 *
+	 * @return PHPUnit_Framework_MockObject_MockObject
+	 */
+	private function mockInfoboxRenderServiceHelper( $input ) {
+		$isValidHeroDataItem = isset( $input[ 'isValidHeroDataItem' ] ) && $input[ 'isValidHeroDataItem' ];
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "isValidHeroDataItem" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)